server/src/main/java/org/apache/lucene/search/grouping/CollapsingTopDocsCollector.java

```diff
@@ -61,11 +65,38 @@ public final class CollapsingTopDocsCollector<T> extends FirstPassGroupingCollec
     protected Scorable scorer;
 
     private int totalHitCount;
+    private final FieldDoc after;
+    private FieldComparator<?> afterComparator;
+    private LeafFieldComparator leafComparator;
+    private final int reverseMul;
 
     CollapsingTopDocsCollector(GroupSelector<T> groupSelector, String collapseField, Sort sort, int topN) {
         super(groupSelector, sort, topN);
         this.collapseField = collapseField;
         this.sort = sort;
+        this.after = null;
+        this.reverseMul = 1;
+    }
+
+    CollapsingTopDocsCollector(GroupSelector<T> groupSelector, String collapseField, Sort sort, int topN, FieldDoc after) {
+        super(groupSelector, sort, topN);
+        this.collapseField = collapseField;
+        this.sort = sort;
+        this.after = after;
+
+        if (after != null) {
+            // we have only one sort field which is the collapse field
```

Comment:
Do we need to enforce that? May let's check if we indeed have only one sorting field and raise exception otherwise

Reply:
Thanks, that makes sense, I've added some validation here.